### PR TITLE
Sogar registry

### PR DESCRIPTION
**Pull request by AnastasiiaRomaniuk**

### CODE REVIEW

File: rust/sogar/sogar-core/src/sogar_registry.rs [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+
+        if let Some(file) = self.file {
+            builder_copy = builder_copy.file(file);
+        }
+
+        builder_copy
+    }
+}
+
+pub struct SogarController;
```
Comment (CBenoit) [inline rust/sogar/sogar-core/src/sogar_registry.rs:74]: Nothing prevents us from building a SogarController without calling `SogarController::new` here.
You can add a zero-sized attribute so we need to call `new` and checks are performed.
```suggestion
pub struct SogarController { _priv: () }
```

**Review by CBenoit — CHANGES_REQUESTED** [review]
Thank you! Here are a few suggestions.

File: rust/sogar/sogar-core/src/sogar_config.rs [hunk 2]
```diff
@@ -119,7 +119,7 @@ pub fn match_arguments(matches: &ArgMatches, config_cache: &mut ConfigCache) ->
     Ok(())
 }
 
-fn get_mime_type_from_file_extension(file_name: String) -> String {
+pub fn get_mime_type_from_file_extension(file_name: String) -> String {
     // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
 
     let mut mime_mapping = HashMap::with_capacity(32);
```
[the remark this answers is not in the document]
Comment (AnastasiiaRomaniuk) [reply rust/sogar/sogar-core/src/sogar_config.rs]: done

File: rust/sogar/sogar-core/src/sogar_registry.rs (hunk 1 above)
Comment (AnastasiiaRomaniuk) [reply rust/sogar/sogar-core/src/sogar_registry.rs]: done

File: rust/sogar/sogar-core/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+pub mod logger;
+mod sogar;
+pub mod sogar_config;
+pub mod sogar_registry;
```
[the remark this answers is not in the document]
Comment (AnastasiiaRomaniuk) [reply rust/sogar/sogar-core/src/lib.rs]: done

**Review by CBenoit — APPROVED** [review]
LGTM. I'll merge after testing.

**Pull request merged**